torchtitan/models/llama/model.py

```diff
@@ -415,8 +415,8 @@ def _precompute_freqs_cis(self) -> torch.Tensor:
         return precompute_freqs_cis(
             self.model_args.dim // self.model_args.n_heads,
             # Need to compute until at least the max token limit for generation
-            # (use 2x max sequence length to be safe)
-            self.model_args.max_seq_len * 2,
+            # Note: removed the 2x relaxing in CP enablement
+            self.model_args.max_seq_len,
```

Comment:
cc., @tianyu-l Want to understand is this okay?

For a general use case, we can also expand the CP to support `stride`-like feature.

Reply:
Could you please elaborate a bit on why this change was needed by CP?

Reply:
@tianyu-l CP parallelize on the sequence dimension, anything related to the sequence dimension needs to be shard. So freqs_cis is the positional embedding and is required to be sharded according to the sequence length. So it is easier to support CP if everything has the same sequence length.